habitat/robots/robot_base.py

```diff
@@ -0,0 +1,176 @@
+                self.params.wheel_mtr_max_impulse,  # max_impulse
+            )
+            # pylint: disable=not-an-iterable
+            for i in self.params.wheel_joints:
+                self.sim_obj.update_joint_motor(self.joint_motors[i][0], jms)
+        self._update_motor_settings_cache()
+
+    def update(self) -> None:
+        pass
+
+    def reset(self) -> None:
+        pass
+
+    #############################################
+    # BASE RELATED
+    #############################################
+    @property
+    def base_pos(self):
+        """Get the robot base ground position"""
+        # via configured local offset from origin
+        if self._base_type == "mobile":
+            return (
+                self.sim_obj.translation
+                + self.sim_obj.transformation.transform_vector(
+                    self.params.base_offset
+                )
+            )
+        elif self._base_type == "static":
+            return mn.Vector3()  # zeros
+        else:
+            raise NotImplementedError("The base type is not implemented.")
+
+    @base_pos.setter
+    def base_pos(self, position: mn.Vector3):
+        """Set the robot base to a desired ground position (e.g. NavMesh point)"""
+        # via configured local offset from origin.
+        if self._base_type == "mobile":
+            if len(position) != 3:
+                raise ValueError("Base position needs to be three dimensions")
+            self.sim_obj.translation = (
+                position
+                - self.sim_obj.transformation.transform_vector(
+                    self.params.base_offset
+                )
+            )
+        # attempts to change the base position will result in an error to prevent accidental mobility if the base is static
+        elif self._base_type == "static":
+            raise NotImplementedError(
+                "Setting the base position of a static manipulator is not permitted."
+            )
+        else:
+            raise NotImplementedError("The base type is not implemented.")
+
+    @property
+    def base_rot(self) -> float:
+        return self.sim_obj.rotation.angle()
+
+    @base_rot.setter
+    def base_rot(self, rotation_y_rad: float):
+        if self._base_type == "mobile":
+            self.sim_obj.rotation = mn.Quaternion.rotation(
+                mn.Rad(rotation_y_rad), mn.Vector3(0, 1, 0)
+            )
+        elif self._base_type == "static":
+            raise NotImplementedError(
+                "Setting the base rotation of a static manipulator is not permitted."
+            )
+        else:
+            raise NotImplementedError("The base type is not implemented.")
+
+    @property
+    def base_transformation(self):
+        return self.sim_obj.transformation
+
+    def is_base_link(self, link_id: int) -> bool:
+        return (
+            self.sim_obj.get_link_name(link_id) in self.params.base_link_names
+        )
+
+    #############################################
```

Comment:
While I like your sections devisions here we don't have it as a standard for other source files. To keep code unified let's avoid it here. Thank you.

Reply:
Done!